text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,811 @@
+      - service.instance.id
+      - service.name
+- [OTEL operator](https://github.com/open-telemetry/opentelemetry-operator/blob/a1e8f927909b81eb368c0483940e0b90d7fdb057/pkg/instrumentation/sdk_test.go#L752) injected ENV variables
+  - service.instance.id
+  - service.name
+  - service.version
+  - k8s.namespace.name
+  - k8s.pod.name
+  - k8s.node.name
+  - k8s.container.name
+
+### Implications
+
+Some initial thoughts on implications:
+
+AWS, Azure, GCP, Heroku, etc. all provide the following "bundles" of resource:
+
+- `cloud.*`
+- `faas.*`, when relevant
+- `host.*`, when relevant
+- `k8s.cluster.*`, when relevant
+- `service.*` when relevant
+- `container.*` for a subset of k8s providers
+
+"system" detection provides the following:
+
+- `host.*`
+- `os.*`
+- `process.*` for SDKs
+- `container.*` for Docker images
+
+SDK specific detection provides the following:
+
+- `sdk.*`
+- `service.*`
+
+The OTEL operator for k8s provides the following via ENV variables:
+
+- `k8s.namespace.*`
+- `k8s.node.*`
+- `k8s.pod.*`
+- `k8s.container.*`
+- `service.*`
+
+### What could this mean for chosing entities that belong on resource?
+
+Let's look at an example of a container running in kubernetes, specifically EKS.
+
+If the OTEL operator, the SDK and the collector are all used, the following
+attributes will wind up on resource:
+
+- `service.*` - from SDK and otel operator
+- `sdk.*` - from SDK
+- `process.*` - from SDK
+- `host.*` - Note: from system detector on collector
+- `container.*` - from EKS detector on SDK
+- `k8s.namespace.*` - from otel operator
+- `k8s.node.*` - from otel operator
+- `k8s.pod.*` - from otel operator
+- `k8s.container.*` - from otel operator
+- `k8s.cluster.*` - from EKS detector on SDK or collector
+- `cloud.*` - from EKS detector on SDK or collector
+
+A simple litmus test derived from this for when to include an entity on
+Resource would be: "Any entity relevant to the produced telemetry should be
+included".
+
+However, this can be refined.  Resources today provide a [few key features](https://docs.google.com/document/d/1Xd1JP7eNhRpdz1RIBLeA1_4UYPRJaouloAYqldCeNSc/edit):
+
+- They provide identity - Uniquely identifying the origin of the data.
+- They provide "navigationality" - allowing users to find the source of the data within their o11y and infrastructure tools.
+- They allow aggregation / slicing of data on interesting domains.
+
+A litmus test for what entities to include on resource should be as follows:
+
+- Is the entity the source/origin of the data?
+- Does the entity help navigate to the source of the data? (e.g. `k8s.cluster.*` helping find a `k8s.container.*`)
+- Do want to easily slice/aggregate on an axis provided by the entity? (e.g. quickly filtering all CPU container usage metrics across a cluster to find overloaded nodes).
+
+If the answer to any question is yes, then include the entity on resource.
```

Comment:
This is quite open-ended. Do you think that limiting a single domain to only include a single most specific detected entity in the resource is an option? And everything else is reported via relations instead of bundling everything together. That way the above example could be reduced to the following entities:

- `service`
- `telemetry.sdk`
- `process`
- `host`
- `container`
- `k8s.pod`
- `cloud.?`

This assumes that we can either somehow distinguish non-prefixed types into relevant domains, i.e. based on 'Schema Family identifier' of the schema_url (to preserve backward compatibility) or we would need to prefix every entity type (i.e. host -> infra.host) for consistency

Reply:
It depends on how we define identity.  With how things work today - We *cannot* limit to just one per domain.  We would need to alter entities in semantic conventions.  E.g. `k8s` related attributes today require *many* entities to really be useful.

E.g.  `k8s.namespace.name`, `k8s.pod.name`, `k8s.cluster.name` <- these are across three "entities" today.

If you have time - I'd like to see a proposal on how to address this with domains to help us answer that open question.  It needs to address what we do about semconv today and how evolve the model.

For the purpose of this proposal - I think we need to spend time addressing domains and entity modelling, but I believe we can do so while prototyping - Hence this remains an open question for us to resolve as we move forwards.